app/src/main/java/ch/epfl/sdp/blindly/viewmodel/UserViewModel.kt

```diff
@@ -50,13 +51,29 @@ class UserViewModel @AssistedInject constructor(
      */
     fun <T> updateField(field: String, newValue: T) {
         viewModelScope.launch {
-            userRepo.updateProfile(userId, field, newValue)
+            userRepository.updateProfile(userId, field, newValue)
+        }
+    }
+
+    fun deleteUser() {
+        viewModelScope.launch {
+            val recordingPath = Recordings.getPresentationAudionName(userId)
+            recordings.deleteFile(recordingPath, object :
+                Recordings.RecordingOperationCallback() {
+                override fun onSuccess() {
+                    Log.d(TAG, "Successfully removed the audio.")
+                }
+                override fun onError() {
+                    Log.e(TAG, "An erro roccured, the audio could not be deleted.")
+                    deleteUser() //Try again
```

Comment:
Well this is dangerous I believe, we are risking a recursive infinite loop in case of a continuous error, if I'm not mistaken. Maybe you can add a simple counter to stop retrying after a couple of attempts.

Reply:
Yep I wasn't so sure about this one I'll remove it!! x)